intake_cesm/manage_collections.py

```diff
@@ -364,8 +371,9 @@ def _build_cesm_collection(self, collection_attrs):
         # write data to csv
         self.df.to_csv(self.active_db, index=True)
 
-    def build_collections(self, overwrite_existing):
+    def _build_collections(self, overwrite_existing):
```

Comment:
@andersy005, it strikes me that this routine is no-longer exactly appropriate in this class. It carries unnecessary logic to select the `collection_type`. I think that logic belongs in `CesmMetadataStoreCatalog.build_collection`. Let's imagine our proposed extension to encompass `intake-cmip`: we will need to understand the `collection_type` to know which method to call at the `CesmMetadataStoreCatalog` level. 

Let me know what you think. I am happy to accept this PR and address this in a future iteration.

Reply:
These are great ideas as we embark on `intake-cmip` and `intake-cesm` consolidation journey. I opened an issue https://github.com/NCAR/intake-cesm/issues/30 for API discussion. Let's iron out the API implementation details for this in that issue. 

For the time being, I am happy with merging this PR the way it is, and addressing the issues you raised in a separate PR.

Reply:
Sounds good.